terraform/modules/ingress/certificates.tf

```diff
@@ -0,0 +1,179 @@
+resource "tls_private_key" "ingress_private_key" {
+  count       = length(tls_private_key.root_ingress)
+  algorithm   = "RSA"
+  ecdsa_curve = "P384"
+  rsa_bits    = "4096"
+}
+
+resource "tls_cert_request" "ingress_cert_request" {
+  count           = length(tls_private_key.ingress_private_key)
+  private_key_pem = tls_private_key.ingress_private_key.0.private_key_pem
+  subject {
+    country     = "France"
+    common_name = "localhost"
+  }
+}
+
+resource "tls_locally_signed_cert" "ingress_certificate" {
+  count                 = length(tls_cert_request.ingress_cert_request)
+  cert_request_pem      = tls_cert_request.ingress_cert_request.0.cert_request_pem
+  ca_private_key_pem    = tls_private_key.root_ingress.0.private_key_pem
+  ca_cert_pem           = tls_self_signed_cert.root_ingress.0.cert_pem
+  validity_period_hours = "168"
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth",
+    "client_auth",
+    "any_extended",
+  ]
+}
+
+#------------------------------------------------------------------------------
+# Client Certificate
+#------------------------------------------------------------------------------
+resource "tls_private_key" "ingress_client_private_key" {
+  count       = length(tls_private_key.client_root_ingress)
+  algorithm   = "RSA"
+  ecdsa_curve = "P384"
+  rsa_bits    = "4096"
+}
+
+resource "tls_cert_request" "ingress_client_cert_request" {
+  count           = length(tls_private_key.ingress_client_private_key)
+  private_key_pem = tls_private_key.ingress_client_private_key.0.private_key_pem
+  subject {
+    country     = "France"
+    common_name = "client"
+    # organization = "127.0.0.1"
+  }
+}
+
+resource "tls_locally_signed_cert" "ingress_client_certificate" {
+  count                 = length(tls_cert_request.ingress_client_cert_request)
+  cert_request_pem      = tls_cert_request.ingress_client_cert_request.0.cert_request_pem
+  ca_private_key_pem    = tls_private_key.client_root_ingress.0.private_key_pem
+  ca_cert_pem           = tls_self_signed_cert.client_root_ingress.0.cert_pem
+  validity_period_hours = "168"
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth",
+    "client_auth",
+    "any_extended",
+  ]
+}
+
+resource "pkcs12_from_pem" "ingress_client_pkcs12" {
+  count           = length(tls_locally_signed_cert.ingress_client_certificate)
+  password        = ""
+  cert_pem        = tls_locally_signed_cert.ingress_client_certificate.0.cert_pem
+  private_key_pem = tls_private_key.ingress_client_private_key.0.private_key_pem
+  ca_pem          = tls_self_signed_cert.client_root_ingress.0.cert_pem
+}
+
+resource "local_sensitive_file" "ingress_ca" {
+  count           = length(tls_self_signed_cert.root_ingress)
+  content         = tls_self_signed_cert.root_ingress.0.cert_pem
+  filename        = "${path.root}/.terraform/${var.container_name}/server/ca.crt"
```

Comment:
Why do you put these in `.terraform`? This folder should only be used by terraform.